lib/_http_server.js

```diff
@@ -313,7 +313,7 @@ function Server(options, requestListener) {
   } else if (options == null || typeof options === 'object') {
     options = { ...options };
   } else {
-    throw new ERR_INVALID_ARG_TYPE('options', 'object', options);
+    throw new ERR_INVALID_ARG_TYPE('options', 'Object', options);
```

Comment:
What's the justification for changing `object` to `Object`? Here's the things that leap to mind as arugments against it:

* `typeof {} === 'object'` and not `'Object'`
* Sure, you can do `new Object()` but you can also do `new String()` or `new Number()` and we don't capitalize those.
* "Everything is lowercased" is an easy rule to remember. "Some things get an initial capital letter and other things don't" is prone to human error.
* No added value to the end user of saying _Object_ instead of _object_.
* Not doing that change will greatly reduce the diff size, making this much easier to review as well as backport. It also reduces the risk of conflicts with other PRs.

Reply:
The justification was that there are already multiple entries written as `Object` and the documentation uses `Object` instead of `object`. I personally actually favor `object` a tiny bit above `Object`. I just roughly remembered that there where comments about this in a PR quite a while ago (I do not remember which one though), so I thought I'll go for that.

This actually also applies to the type `function`. It is mostly written as `Function`.

@vsemozhetbyt I think you where involved in this discussion. Do you by any chance remember any details?

Reply:
> @Trott I am fine to convert all `Function` and `Object` entries to lower cased versions, if you think that's a good way to handle it. I personally favor that.

I'm fine either way.

Reply:
I had another look at it and there is a good reason to keep `Object` instead of `object`: when talking about the `type` I expect it to correspond to `typeof input === 'object'`. When `Object` is used I would expect the direct prototype to be `Object` compared to e.g., `RegExp` or `Array` or any other input if type `object`. Otherwise it's weird to have entries like:

`The "error" argument must be one of type object or function or an instance of Error, or RegExp`.
An instance of `Error` or `RegExp` is always of type `object`. So it's "redundant". But removing the "redundancy" would limit the usefulness of the output significantly.

Reply:
> when talking about the `type` I expect it to correspond to `typeof input === 'object'`. When `Object` is used I would expect the direct prototype to be `Object

I might be misunderstanding you, but those seem like arguments for keeping lowercase `object` here, because it *is* about `typeof` and not the prototype constructor name?

Reply:
It's correct that we use `typeof input === 'object'` in (probably) all of these cases. But when we do, we normally mean the object should have `Object` as prototype (or a null prototype as in `Object.create(null)`).

The example above is more intuitive for me if it's written as:
`The "error" argument must be of type function or an instance of Object, Error, or RegExp`. That way I would immediately expect the input to be allowed to look like:

```js
let input = {} // Valid
input = /a/ // Valid
input = new TypeError() // Valid
input = new Map() // Invalid
```

Reply:
@addaleax what do you think would be best here? The PR should overall reduce ambiguity and seems clearer than before.

Reply:
> But when we do, we normally mean the object should have `Object` as prototype (or a null prototype as in `Object.create(null)`).

I don’t think that’s true, and the way that JS objects work, the `Map` instance in your example *is* valid because we can attach regular properties to it, too.

So I’d personally stick with the current formatting of lowercase `object` unless we actually perform some kind of `instanceof`-style check.

Reply:
You are correct that it would be possible to use any object if we do the typecheck only. What we normally do is a check for a specific class, than another one and in the end we accept any object as long as it seems to fit.

However, we do mostly not accept `null` and that is also of type `object`.

I decided to go for a middle ground and hope that is good for everyone: if no class is detected, we'll always use `type` in combination with `object`. If we detect a class and `object` as accepted values, we'll declare it as  ìnstance of Object`. This condition is pretty rare and I could only find few errors where this would apply too. That way we'll keep the distinction between different object types and use `type of object` in almost all other cases.

Is that fine for you?